File: exmplos/py/coesao_desacoplamento/src/pai.py
from src.abstrata import Abstrata
from src.pessoas import TipoDePessoa


class Pai(TipoDePessoa, Abstrata):

    lista = []

    def __init__(self, cpf='', cnpj='', data_nascimento='', nome='', sobrenome=''):
        super(Pai, self).__init__(cpf, cnpj, data_nascimento, nome, sobrenome)

    def cont_0_9(self, a):
        self.lista.append(a)
        while len(self.lista) < 10:
            return self.cont_0_9(len(self.lista))

File: exmplos/java/coesao_desacoplamento/oo/Pai.java
package oo;

public class Pai extends TipoPessoa implements Abstrata {

    public Pai(){}

    public Pai(String cpf, String cnpj,
               String data_nescimento,
               String nome,
               String sobrenome){
        super.setCpf(cpf);
        super.setCnpj(cnpj);
        super.setData_nascimento(data_nescimento);
        super.setNome(nome);
        super.setSobrenome(sobrenome);
    }

    int[] lista = new int[10];

    public int cont_0_9(int a){
        for(int i =0; i < 10; i++) {
            lista[i] = a;
            return lista[i];
        }
        return 0;
    }

}